solidity/contracts/libs/Call.sol

```diff
@@ -9,43 +9,78 @@ library CallLib {
         bytes data;
     }
 
-    function _multicall(address to, bytes[] memory calls) internal {
+    struct CallWithValue {
+        uint256 value;
+        Call _call;
+    }
+
+    struct CallWithCallback {
+        Call _call;
+        bytes callbackdata;
+    }
+
+    function _call(Call memory call) internal returns (bytes memory) {
+        return Address.functionCall(call.to, call.data);
+    }
+
+    function _call(CallWithValue memory call) internal returns (bytes memory) {
+        return
+            Address.functionCallWithValue(
+                call._call.to,
+                call._call.data,
+                call.value
+            );
+    }
+
+    function _call(CallWithCallback memory call)
+        internal
+        returns (bytes memory)
+    {
+        return bytes.concat(call.callbackdata, _call(call._call));
+    }
+
+    function multicall(Call[] memory calls) internal {
         uint256 i = 0;
         uint256 len = calls.length;
         while (i < len) {
-            Address.functionCall(to, calls[i]);
+            _call(calls[i]);
             unchecked {
                 ++i;
             }
         }
     }
 
-    function _multicall(Call[] memory calls) internal {
+    function multicall(CallWithValue[] memory calls) internal {
         uint256 i = 0;
         uint256 len = calls.length;
         while (i < len) {
-            Address.functionCall(calls[i].to, calls[i].data);
+            _call(calls[i]);
             unchecked {
                 ++i;
             }
         }
     }
 
-    function _multicallAndResolve(Call[] memory calls, bytes[] memory callbacks)
+    function multicall(CallWithCallback[] memory calls)
         internal
-        returns (bytes[] memory resolveCalls)
+        returns (bytes[] memory callbacks)
```

Comment:
It might be less gas efficient but having this return `Call[] memory cbs` would be more readable, IMO

Reply:
agreed more readable